### PR TITLE
Integrate new diagram-js canvas - focus on canvas after editing

### PR DESCRIPTION
**Pull request by jarekdanielak**
Related to https://github.com/bpmn-io/diagram-js/issues/661

### Proposed Changes

Refocus back on canvas on close. 

 

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

**Comment by jarekdanielak** [comment]
@nikku this is now ready for review since `diagram-js@15.0.0` is released.

File: lib/DirectEditing.js [hunk 1]
```diff
@@ -78,6 +80,8 @@ DirectEditing.prototype.close = function() {
   this._active = null;
 
   this.resizable = undefined;
+
+  this._canvas.restoreFocus();
```
Comment (nikku) [inline lib/DirectEditing.js:84]: Consider the upgrade path. This breaks with `diagram-js <= 15`.

**Review by nikku — CHANGES_REQUESTED** [review]
Not backwards compatible. Fine if we release this as a major.

However, as we mark `diagram-js: *` [as a peer dependency](https://github.com/bpmn-io/diagram-js-direct-editing/pull/49/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R50) I'd rather gracefully handle `Canvas#restoreFocus` not being there.

**Comment by jarekdanielak** [comment]
Great point about backwards compatibility.

I added a safety check.

File: lib/DirectEditing.js [hunk 2]
```diff
@@ -78,6 +80,9 @@ DirectEditing.prototype.close = function() {
   this._active = null;
 
   this.resizable = undefined;
+
+  // restoreFocus API is available from diagram-js@15.0.0
```
Comment (nikku) [inline lib/DirectEditing.js:84]: What about adding an integration test, verifying this continues to work with diagram-js@14?

Comment (nikku) [reply lib/DirectEditing.js:84]: Playbook here: https://github.com/bpmn-io/bpmn-js-properties-panel/blob/main/.github/workflows/CI.yml

(Basically execute the tests twice, once with normal dependency, once against defined diagram-js version).

**Review by nikku — COMMENTED** [review]
@jarekdanielak Not sure how you ended up with your integration CI pattern (cf. [reference I used](https://github.com/bpmn-io/bpmn-js-token-simulation/blob/main/.github/workflows/CI.yml)).

[This should do the trick](https://github.com/bpmn-io/diagram-js-direct-editing/pull/49/commits/c0f2ea12297caa1d3fa79f14cf378249e8d61c3a) to test against different set of integration deps, cf. https://github.com/bpmn-io/diagram-js-direct-editing/actions/runs/11701527355/job/32587815880?pr=49

Let's squash the CI changes, rest looks good to me to merge.

**Comment by nikku** [comment]
Follow-up fix: https://github.com/bpmn-io/diagram-js-direct-editing/commit/77114771245eca30ff19b62aa44f4fb1dd62ae17.

**Pull request merged**